prometheus/internal/go_runtime_metrics.go

```diff
@@ -75,3 +76,49 @@ func RuntimeMetricsToProm(d *metrics.Description) (string, string, string, bool)
 	}
 	return namespace, subsystem, name, valid
 }
+
+// RuntimeMetricsBucketsForUnit takes a set of buckets obtained for a runtime/metrics histogram
+// type (so, lower-bound inclusive) and a unit from a runtime/metrics name, and produces
+// a reduced set of buckets. This function always removes any -Inf bucket as it's represented
+// as the bottom-most upper-bound inclusive bucket in Prometheus.
+func RuntimeMetricsBucketsForUnit(buckets []float64, unit string) []float64 {
+	switch unit {
+	case "bytes":
+		// Rebucket as powers of 2.
+		return rebucketExp(buckets, 2)
+	case "seconds":
+		// Rebucket as powers of 10 and then truncate all buckets
+		// above 1 second and merge them into a single bucket.
+		b := rebucketExp(buckets, 10)
+		for i := range b {
+			if b[i] > 1 {
```

Comment:
nit: We could eliminate <= 1 first to keep code more left indent

Reply:
done